R/geo_objects.R

```diff
@@ -0,0 +1,56 @@
+# Contains functions to parse the objects described here: 
+# https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/geo
+
+bounding_box <- function(x) {
+  empty <- data.frame(long = NA, lat = NA, type = NA)
+  if(is.null(x) || length(x) == 1 && is.na(x)) {
+    return(empty)
+  }
+  if (is.data.frame(x)) {
+    coord <- x$coordinates[[1]][1, , ]
+    if (is.null(coord)) {
+      return(empty)
+    }
+    return(data.frame(long = coord[, 1], lat = coord[, 2], type = I(x$type)))
+  }
+  m <- x$coordinates[1, , ]
+  colnames(m) <- c("long", "lat")
+  df <- as.data.frame(m)
+  df$type <- I(x$type)
```

Comment:
You only need to use `I()` when you're creating a list-column with `data.frame()`.

Reply:
Somehow I thought it was a data.frame. Removed.